token.go

```diff
@@ -850,6 +897,10 @@ func (t tokenProcessor) nextToken() (tokenStruct, error) {
 			return nil, nil
 		}
 	case <-t.ctx.Done():
+		// It seems the Message function on t.outs.msgq doesn't get the Done if it comes here instead
```

Comment:
this case seems to be somewhat problematic. Occasionally the new test or benchmark will fail because it only reads 3 result sets instead of 4. The common behavior in each of the failures is that the log has an extra DoneStruct with status = 32 indicating we went down this code path without reading the row tokens from the channel first.
Do you have any clues on what might causes ctx.Done() to fire here before we empty the channel? Should processSingleResponse not return right away when it gets a doneStruct with an even numbered status? I am going to add some logging locally to try to track this down.